storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -67,6 +73,12 @@ public TopologyDetails(String topologyId, Map topologyConf, StormTopology topolo
         this.initConfigs();
     }
 
+    public TopologyDetails(String topologyId, Map topologyConf, StormTopology topology,
+                           int numWorkers, Map<ExecutorDetails, String> executorToComponents, int launchTime) {
+        this(topologyId, topologyConf, topology, numWorkers, executorToComponents);
+        this.launchTime = launchTime;
+    }
```

Comment:
It would be better to have the less specialized constructor call into the more specialized constructor, instead of the other way around.